### PR TITLE
fix: adding readiness probe annotations for openfaas

### PR DESCRIPTION
**Pull request by abhimanyubabbar**
## What are the changes introduced in this PR?

Adding readiness annotations for the python openfaas transformations which allows the endpoints for the python transformations to be registered with kubernetes once we are ready which includes downloading the source code for the transformations.

## What is the related Linear task?
Fixes DAT-1385

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3529/test-report.html

**Review by kanishkkatara — DISMISSED** [review]

File: src/util/openfaas/index.js [hunk 1]
```diff
@@ -27,6 +27,14 @@ const FAAS_LIMITS_MEMORY = process.env.FAAS_LIMITS_MEMORY || FAAS_REQUESTS_MEMOR
 const FAAS_MAX_INFLIGHT = process.env.FAAS_MAX_INFLIGHT || '4';
 const FAAS_EXEC_TIMEOUT = process.env.FAAS_EXEC_TIMEOUT || '4s';
 const FAAS_ENABLE_WATCHDOG_ENV_VARS = process.env.FAAS_ENABLE_WATCHDOG_ENV_VARS || 'true';
+const FAAS_READINESS_HTTP_PATH = process.env.FAAS_READINESS_HTTP_PATH || '/ready';
+const FAAS_READINESS_HTTP_INITIAL_DELAY_S = process.env.FAAS_READINESS_HTTP_INITIAL_DELAY_S || '1';
```
Comment (Jayachand) [inline src/util/openfaas/index.js:31]: should we have intial delay as 5sec. given pod needs to load transformations and libraries

Comment (abhimanyubabbar) [reply src/util/openfaas/index.js:31]: For smaller transformations 1sec would be enough so they won't need to wait for the probe to start for so long

Comment (abhimanyubabbar) [reply src/util/openfaas/index.js:31]: I can increase the threshold of failure to 5 maybe so all in all maybe 11s to download the libraries and transformations

**Review by kanishkkatara — APPROVED** [review]

**Review by Jayachand — APPROVED** [review]

**Pull request merged**